algebra/src/polynomial/field/coeff/mul.rs

```diff
@@ -0,0 +1,58 @@
+use crate::{
+    modulus::ShoupFactor,
+    reduce::{ReduceAddAssign, ReduceMul, ReduceMulAdd, ReduceMulAssign},
+    Field,
+};
+
+use super::FieldPolynomial;
+
```

Comment:
We also need a polynomial mul function here? It may be easier to directly invoke this function without explicitly convert it into the ntt version first.